extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -95,13 +113,25 @@
   private final InputRowParser<ByteBuffer> parser;
   private final KafkaTuningConfig tuningConfig;
   private final KafkaIOConfig ioConfig;
+  private final Optional<ChatHandlerProvider> chatHandlerProvider;
+
+  private final Map<Integer, Long> endOffsets = new ConcurrentHashMap<>();
+  private final Map<Integer, Long> nextOffsets = new ConcurrentHashMap<>();
+  private final Lock pauseLock = new ReentrantLock();
```

Comment:
Could you include a comment about what pauseLock is meant to be synchronizing and what shouldPause/shouldResume should be signaling?